File: src/Python/7.整数反转.py
#
# @lc app=leetcode.cn id=7 lang=python3
#
# [7] 整数反转
#
class Solution:
    def reverse(self, x: int) -> int:
        sign = 1 if x > 0 else -1
        res = 0
        x = abs(x)
        while x :
            res = res*10 + x%10
            if res > 2**31 - 1:
                return 0
            x = x//10

        return sign * res

File: src/Cplusplus/7.整数反转.cpp
/*
 * @lc app=leetcode.cn id=7 lang=cpp
 *
 * [7] 整数反转
 */
#include <climits>
#include <iostream>

class Solution {
public:
    int reverse(int x) {
        long result = 0;
        while(x != 0)
        {
            result = result*10 + x % 10;
            std::cout << result;
            x /= 10;
        }
        return (result > INT_MAX || result < INT_MIN)? 0 : result;
        }
};
/*
int main()
{
    Solution s;
    std::cout << s.reverse(123) << std::endl;
    std::cout << s.reverse(-123) << std::endl;
    std::cout << s.reverse(10100) << std::endl;
    std::cout << s.reverse(1000000003) << std::endl;

    return 0;
}
*/
